applications/tari_console_wallet/src/lib.rs

```diff
@@ -213,16 +212,7 @@ fn get_password(config: &ApplicationConfig, cli: &Cli) -> Option<SafePassword> {
 
 fn get_recovery_seed(boot_mode: WalletBoot, cli: &Cli) -> Result<Option<CipherSeed>, ExitError> {
     if matches!(boot_mode, WalletBoot::Recovery) {
-        let seed = if cli.seed_words.is_some() {
-            // need to zeroize first, to clean up memory of cli.seed_words clone
-            let seed_words: SeedWords = SeedWords::new(
-                cli.seed_words
-                    .as_ref()
-                    .unwrap()
-                    .split_whitespace()
-                    .map(|s| Hidden::hide(s.to_string()))
-                    .collect(),
-            );
+        let seed = if let Some(seed_words) = cli.seed_words.clone() {
             get_seed_from_seed_words(seed_words)?
```

Comment:
nit: get_seed_from_seed_words can take a reference, then you don't need to clone the secret value (generally a good thing but still a nit)
```suggestion
        let seed = if let Some(ref seed_words) = cli.seed_words {
            get_seed_from_seed_words(seed_words)?
```

Reply:
Absolutely ! Thanks for the suggestion ! :)